pkg/storage/postgres/postgres_test.go

```diff
@@ -26,6 +28,30 @@ func TestPostgresDatastore(t *testing.T) {
 	test.RunAllTests(t, ds)
 }
 
+func TestMigrate(t *testing.T) {
+	engine := "postgres"
+	// starts the container and runs migration up to the latest migration version available
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, engine)
+
+	uri := testDatastore.GetConnectionURI()
+	ds, err := New(uri, sqlcommon.NewConfig())
+	require.NoError(t, err)
+	defer ds.Close()
+
+	// going from version 3 to 4 when migration #4 doesn't exist is a no-op
+	version := testDatastore.GetDatabaseVersion() + 1
```

Comment:
This behavior and comment is a bit misleading. If you try to migrate to revision 4 and it is an undefined revision, then that should return an error. There should be two cases worth testing

1. Omitting the revision defaults to latest.

   `openfga migrate --datastore-engine=postgres --datastore-uri="..."` (defaults to latest)

2. Specifying the revision explicitly should require a valid revision number.

    `openfga migrate --datastore-engine=postgres --datastore-uri="..." 10` (error, revision 10 is undefined)

    `openfga migrate --datastore-engine=postgres --datastore-uri="..." 3` (success)

Reply:
> If you try to migrate to revision 4 and it is an undefined revision, then that should return an error.

If you're on version 3 and try to migrate to version 4, Goose doesn't do anything. It feels weird to introduce an error when there is none, I see it as a deletion where it's already deleted, in REST that results in a no-op as well

Reply:
@jon-whit i'm going to merge this as-is but if you think we should introduce an error when users try to migrate from version X to version X+1 when the highest one available is X, please open an issue. Thanks!